hangman.py

```diff
@@ -0,0 +1,311 @@
+    '''
+    warnings = 3
+    attemps = 6
+    letters_guessed = set()
+    while True:
+        print("_" * 16)
+        print("Warnings: ", warnings)
+        print("Available letters: ",get_available_letters(letters_guessed))
+        print("Attemps: ", attemps)
+        letter = input("enter letter:").strip(' ')
+        letter.lower()
+        if not letter.isalpha() or letter in letters_guessed:
+            if warnings >0:
+                warnings -=1
+                print("Oooops! It`s a danger!!!")
+            else:
+                attemps -=1
+        else:
+            letters_guessed.add(letter)
+            splited_word = get_guessed_word(secret_word, letters_guessed)
+            print(splited_word)
+            guessed = splited_word
+            if is_word_guessed(secret_word,letters_guessed):
+                print("You win!!!")
+                break
+            if letter not in secret_word:
+                if letter in "aeoiu":
+                    attemps -= 1
+                attemps -= 1
+        if attemps <= 0:
+            print("You lose!")
+            break
+
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+
+
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+#(hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    my_word_1 = my_word.replace(" ","")
+    if len(my_word_1) != len(other_word):
+        return False
+    for i in range(len(my_word_1)):
+        if my_word_1[i] != "_" and (other_word[i] != my_word_1[i] or my_word_1.count(my_word_1[i]) != other_word.count(other_word[i])):
+            return False
+    return True
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+
+
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+    '''
+    words = []
+    for word in wordlist:
+        if match_with_gaps(my_word, word):
+           words.append(word)
+    if len(words) == 0:
+        print("No matches found!")
+    else:
+        print(str(words)[1:-1])
```

Comment:
Try tp find different list to string convertion, so that you don't need to remove first and last characters